server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOTest.java

```diff
@@ -104,5 +108,23 @@ void failsWithRuntimeExceptionOnCreateContainerTwice() throws Exception {
             .hasCauseInstanceOf(ObjectStoreException.class)
             .hasMessageContaining("Unable to create container");
     }
+
+    @Test
+    void supportsEncryptionWithCustomPayloadCodec() throws Exception {
+        ObjectStorageBlobsDAO encryptedDao = ObjectStorageBlobsDAO
+            .builder(testConfig)
+            .container(containerName)
+            .blobIdFactory(blobIdFactory())
+            .payloadCodec(new AESPayloadCodec(new CryptoConfig(
+                "c603a7327ee3dcbc031d8d34b1096c605feca5e1", "foobar")))
+            .build();
+        byte[] bytes = "James is the best!".getBytes(StandardCharsets.UTF_8);
+        BlobId blobId = encryptedDao.save(bytes).join();
+
+        InputStream read = testee.read(blobId);
+        assertThat(read).isNotNull();
+        byte[] encryptedBytes = IOUtils.toByteArray(read);
+        assertThat(encryptedBytes).isNotEqualTo(bytes);
+    }
```

Comment:
Can we get a save then read test on the encrypted DAO to check that it is still the same payload?